localstack/services/apigateway/apigateway_starter.py

```diff
@@ -117,6 +119,25 @@ def apigateway_response_restapis_individual(self, request, full_url, headers):
         self.setup_class(request, full_url, headers)
         function_id = self.path.replace('/restapis/', '', 1).split('/')[0]
 
+        if self.method == 'PATCH':
+            not_supported_attributes = ['/id', '/region_name', '/create_date']
+
+            rest_api = self.backend.apis.get(function_id)
+            if not rest_api:
+                msg = 'Invalid API identifier specified %s:%s' % (TEST_AWS_ACCOUNT_ID, function_id)
+                return (404, {}, msg)
```

Comment:
The response we are getting out of this is 

- An error occurred `(404)` when calling the UpdateRestApi operation: Invalid API identifier specified 000000000000:930yfo6fv1

instead of 
- An error occurred `(NotFoundException)` when calling the UpdateRestApi operation: Invalid API identifier specified 074255357339:930yfo6fv1